app/models/framework/definition.rb

```diff
@@ -47,7 +47,27 @@ def calculate_management_charge(entry)
         end
 
         def for_entry_type(entry_type)
-          entry_type == 'invoice' ? self::Invoice : self::Order
+          case entry_type.downcase
+          when 'invoice' then self::Invoice
+          when 'order' then self::Order
+          when 'other' then self::Other
+          else
+            raise ArgumentError, "Unknown entry_type of #{entry_type}"
+          end
+        end
```

Comment:
Since we've added `Framework.defined?` we can (should?) use that instead of the `case`?